hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcClient.java

```diff
@@ -368,8 +370,20 @@ private ConnectionContext getConnection(UserGroupInformation ugi, String nsId,
         connUGI = UserGroupInformation.createProxyUser(
             ugi.getUserName(), routerUser);
       }
+
```

Comment:
Couldn't we move this code inside of ConnectionManager.getConnection and avoid changing the API to add the clientStateId?

Reply:
Good catch. I've refactored the code.